src/DataStructures/Tensor/Expressions/TensorIndexTransformation.hpp

```diff
@@ -151,11 +151,10 @@ transform_multi_index(
       make_array<NumIndicesOut, size_t>(0);
   for (size_t i = 0; i < NumIndicesOut; i++) {
     gsl::at(output_multi_index, i) =
-        (gsl::at(tensorindex_transformation, i) ==
+        (gsl::at(tensorindex_transformation, i) !=
```

Comment:
how does this optimization work? Add a code comment?

Reply:
I added a comment on this. Note that I didn't benchmark before vs after this. I just made this swap because it's my understanding that with control flow, in general, we want to put the more likely cases as the `if` instead of the `else`. I don't know if it has a meaningful impact, as this is one of the many small changes I made in the overall redesign. If you prefer I not make this change without benchmarking, let me know and I will remove this change for now!

Reply:
I have no idea if this is faster, but I don't mind the change 🤷‍♂️